basic_module/distribution/src/impls.rs

```diff
@@ -112,3 +112,71 @@ impl FeeManager for Distribution {
         context.set(get_storage_id(), &get_key(), intermediate_rewards.into());
     }
 }
+
+impl TransactionHandler for Distribution {
+    fn create_distribution_transaction(&self) -> Transaction {
```

Comment:
What do you think of changing the type of the transaction `DstributeRewards` more descriptive? The original intention of  introducing such explicit `DistributeRewards` transaction is to make state changes more visible. Though all state transition rules can not be expressed just by its type, we are able to improve it. If the transaction type contains only the map from accounts to distributed rewards then the state tracking can be more easier.
By complicating creation logic, you can simplify its execution.